### PR TITLE
GCI Commits for different tasks

### PR DESCRIPTION
**Pull request by anurag-ks**
"implemented mulitsheet xls export" was for my previous task (which is completed)- https://www.google-melange.com/gci/task/view/google/gci2014/5772250456260608

"Bulk Representation for project_beneficary" is for my current task - https://www.google-melange.com/gci/task/view/google/gci2014/5322710255140864 , Which needs to be reviewed.

### CODE REVIEW

**Comment by anurag-ks** [comment]
Can someone please review my work.

File: modules/s3db/project.py [hunk 1]
```diff
@@ -2002,6 +1975,68 @@ def project_beneficiary_activity_type_deduplicate(item):
...
+    def represent_row(self, row):
+        """
+            Represent a single Row
+
+            @param row: the project_benficiary Row
+        """
+        
+        id = row.id
+        if not id:
+            return current.messagesp["None"]
```
Comment (hitesh96db) [inline modules/s3db/project.py]: Typo

File: modules/s3db/project.py [hunk 2]
```diff
@@ -2002,6 +1975,68 @@ def project_beneficiary_activity_type_deduplicate(item):
...
+
+        try:
+            return "%s %s" % (row["project_benficiary.value"],
```
Comment (hitesh96db) [inline modules/s3db/project.py]: Typo

File: modules/s3db/project.py [hunk 3]
```diff
@@ -2002,6 +1975,68 @@ def project_beneficiary_activity_type_deduplicate(item):
...
+    # -------------------------------------------------------------------------
+    def represent_row(self, row):
+        """
+            Represent a single Row
+
+            @param row: the project_benficiary Row
+        """
+        
+        id = row.id
+        if not id:
```
Comment (hitesh96db) [inline modules/s3db/project.py]: No need of this ? 
Row will always have an id( check this )

File: modules/s3db/project.py [hunk 4]
```diff
@@ -2002,6 +1975,68 @@ def project_beneficiary_activity_type_deduplicate(item):
...
+        left = [ttable.on(table.parameter_id == ttable.id)]
+
+        if len(values) == 1:
+            query = (table.id == values[0])
+            limitby = (0, 1)
+        else:
+            query = (table.id.belongs(values))
+            limitby = (0, len(values))
+
+        rows = db(query).select(*self.fields,
```
Comment (hitesh96db) [inline modules/s3db/project.py]: Bad Syntax. Place self.fields at the end of the params list

File: modules/s3db/project.py [hunk 5]
```diff
@@ -2002,6 +1975,68 @@ def project_beneficiary_activity_type_deduplicate(item):
...
+        self.lookup_rows = self.custom_lookup_rows
+        fields = ["project_beneficiary.value",
+                  "project_beneficiary_type.name"]
+
+        super(project_BeneficiaryRepresent,
+              self).__init__(lookup="project_beneficiary",
+                             fields=fields)
+
+    # -------------------------------------------------------------------------
+    def custom_lookup_rows(self, key, values, fields=[]):
```
Comment (hitesh96db) [inline modules/s3db/project.py]: Change this to lookup_rows and then there's no need of an init function

File: modules/s3db/project.py [hunk 6]
```diff
@@ -1827,15 +1827,15 @@ def model(self):
                   report_options = report_options,
                   super_entity = "stats_data",
                   )
-
+        project_beneficiary_represent = project_BeneficiaryRepresent()
```
Comment (hitesh96db) [inline modules/s3db/project.py]: You'll have to set the lookup here.

**Comment by hitesh96db** [comment]
the Migration to S3Represent looks fine other than those small things.

File: modules/s3db/project.py [hunk 7]
```diff
@@ -2002,6 +1978,57 @@ def project_beneficiary_activity_type_deduplicate(item):
...
+
+    # -------------------------------------------------------------------------
+    def represent_row(self, row):
+        """
+            Represent a single Row
+
+            @param row: the project_benficiary Row
+        """
+        
+        #id = row.id
```
Comment (hitesh96db) [inline modules/s3db/project.py]: Remove these ?

File: modules/s3db/project.py [hunk 8]
```diff
@@ -2002,6 +1978,57 @@ def project_beneficiary_activity_type_deduplicate(item):
...
+        #if not id:
+        #    return current.messages["None"]
+
+        try:
+            return "%s %s" % (row["project_beneficiary.value"],
```
Comment (hitesh96db) [inline modules/s3db/project.py]: Won't work, the rows will be of this form

<pre>row = {"table1": {"field1": "value"},
                    "table2": {"field2": "value"}
                    }
</pre>

File: modules/s3db/project.py [hunk 9]
```diff
@@ -1827,15 +1827,18 @@ def model(self):
                   report_options = report_options,
                   super_entity = "stats_data",
                   )
-
+        project_beneficiary_represent = project_BeneficiaryRepresent(lookup=tablename,
+                                                                     fields = ["id",
```
Comment (hitesh96db) [inline modules/s3db/project.py]: project_beneficiary.id ? I think it's better to put these fields in lookup_rows.

File: tests/implementation/testsuites/smoke_tests/smoke_tests.txt [hunk 10]
```diff
@@ -8,7 +8,7 @@
 # It contains the configuration settings to run the smoke tests
 
 # Setting the depth till where URLs will be parsed
-${MAXDEPTH}  2
+${MAXDEPTH}  5
```
Comment (hitesh96db) [inline tests/implementation/testsuites/smoke_tests/smoke_tests.txt]: ?

**Comment by anurag-ks** [comment]
Okay I have tested, I hope it's good now.

**Comment by anurag-ks** [comment]
I wanted to ask, should i send a new PR(after this get's merged) for that Bug fix for xsl export ?

**Comment by hitesh96db** [comment]
Yes, but did you inform/comment on the Bug Tracker that you are working on it ?

**Comment by anurag-ks** [comment]
Okay I'll add a comment over there, and is the migration correct(If yes please mark my task) ?

**Comment by hitesh96db** [comment]
Not yet, once I do it, I'll inform you.

File: modules/s3db/project.py [hunk 11]
```diff
@@ -2002,6 +1975,55 @@ def project_beneficiary_activity_type_deduplicate(item):
                 item.method = item.METHOD.UPDATE
 
 # =============================================================================
+class project_BeneficiaryRepresent(S3Represent):
+    """ Representation of Organisations """
```
Comment (hitesh96db) [inline modules/s3db/project.py]: ;)

File: modules/s3db/project.py [hunk 12]
```diff
@@ -2002,6 +1975,55 @@ def project_beneficiary_activity_type_deduplicate(item):
...
+        db = current.db
+        table = db.project_beneficiary
+        ttable = db.project_beneficiary_type
+        fields = ["project_beneficiary.id",
+                  "project_beneficiary.value",
+                  "project_beneficiary_type.name"]
+
+        left = [ttable.on(table.parameter_id == ttable.id)]
+
+        if len(values) == 1:
```
Comment (hitesh96db) [inline modules/s3db/project.py]: Better to save this in a variable so that it's only called once.

**Comment by hitesh96db** [comment]
Could you put this class along with other Represent classes in the same file. 
Some minor changes, otherwise :+1:

**Comment by anurag-ks** [comment]
@hitesh96db I have done the changes, If all are correct please mark my task so that I can move forward.
Thanks.

**Comment by anurag-ks** [comment]
I have moved this PR to the staging branch over here - https://github.com/edengci/eden/pull/18

**Pull request closed without merging**